### PR TITLE
Update ChangeLog as preparation for DOMjudge 8.3

### PR DESCRIPTION
**Pull request by vmcj**

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -29,6 +29,21 @@ Version 8.3.0DEV
  - Add direct button to public page for registering when enabled.
  - Scale batch size of judge tasks dynamically.
  - Rename room to location for teams.
+ - Add option to upload contest text/statement.
```
Comment (nickygerritsen) [inline ChangeLog:32]:
```suggestion
 - Add option to upload contest problemset document.
```

File: ChangeLog [hunk 2]
```diff
@@ -29,6 +29,21 @@ Version 8.3.0DEV
...
+ - Improve on the advice for rejuding on config changes, only when relevant
+   advice to start a rejudging, similar for updating the scoreboard cache.
+ - For interactive problems: Add option to upload contest text/statement.
```
Comment (nickygerritsen) [inline ChangeLog:35]: No?

File: ChangeLog [hunk 3]
```diff
@@ -29,6 +29,21 @@ Version 8.3.0DEV
  - Add direct button to public page for registering when enabled.
  - Scale batch size of judge tasks dynamically.
  - Rename room to location for teams.
+ - Add option to upload contest problemset document.
+ - Improve on the advice for rejuding on config changes, only when relevant
+   advice to start a rejudging, similar for updating the scoreboard cache.
+ - For interactive problems: Add option to upload contest text/statement.
```
Comment (vmcj) [inline ChangeLog:35]:
```suggestion
 - For interactive problems: add explicit statement to when one of sides of the interactions ran closed their output.
```

File: ChangeLog [hunk 4]
```diff
@@ -29,6 +29,21 @@ Version 8.3.0DEV
...
+ - Allow importing a (generic) contest warning which is displayed on the scoreboard.
+ - [UI] Use entity badges on submission pages.
```
Comment (meisterT) [inline ChangeLog:37]: This is really minor and we made a dozen of other small UI changes as well. Do we want to list all of them?

Comment (vmcj) [reply ChangeLog:37]: > This is really minor and we made a dozen of other small UI changes as well. Do we want to list all of them?

no, i was already against this one but we had so many that I found it was reasonable to summarize to this one. I'll remove it.

File: ChangeLog [hunk 5]
```diff
@@ -29,6 +29,21 @@ Version 8.3.0DEV
  - Add direct button to public page for registering when enabled.
  - Scale batch size of judge tasks dynamically.
  - Rename room to location for teams.
+ - Add option to upload contest problemset document.
+ - Improve on the advice for rejuding on config changes, only when relevant
+   advice to start a rejudging, similar for updating the scoreboard cache.
+ - For interactive problems: add explicit statement to when one of sides of the interactions ran closed their output.
+ - Allow importing a (generic) contest warning which is displayed on the scoreboard.
+ - [UI] Use entity badges on submission pages.
```
Comment (vmcj) [inline ChangeLog:37]:
```suggestion
```

File: ChangeLog [hunk 6]
```diff
@@ -29,6 +29,33 @@ Version 8.3.0DEV
...
+   easier.
+ - Set default C/C++ standards in configure script, we intend to ship all
+   scripts free of warnings/errors.
+ - Add option to not show any rank on mirror contests, this is useful when
+   you want to disclose the problems but not make comparing easy between the
+   official contest and the mirror.
+ - Allow basic authentication for the import-contest URL.
+ - Fix CORS header for API to make extending with other tools easier.
+ - Fix eventfeed when SQL commits happen out of order to send all events.
+ - Add option to print (on paper) all clarifications.
```
Comment (nickygerritsen) [inline ChangeLog:54]: We already had the option, we now just show the button in a different place.

File: ChangeLog [hunk 7]
```diff
@@ -29,6 +29,33 @@ Version 8.3.0DEV
...
+   colour balloon.
+ - Compile executables with `-std=gnu++20` now.
+ - Record the testcase directory for judging runs, this makes investigating
+   easier.
+ - Set default C/C++ standards in configure script, we intend to ship all
+   scripts free of warnings/errors.
+ - Add option to not show any rank on mirror contests, this is useful when
+   you want to disclose the problems but not make comparing easy between the
+   official contest and the mirror.
+ - Allow basic authentication for the import-contest URL.
```
Comment (nickygerritsen) [inline ChangeLog:51]: This was just fixing that if you had a netrc the script still works with username + password in the URL right? In that case this wording is not very clear.

File: ChangeLog [hunk 8]
```diff
@@ -29,6 +29,33 @@ Version 8.3.0DEV
...
+   upgrades and during the contest before problem imports.
+ - Fixed some bugs to become OpenAPI spec compliant again.
+ - Prevent a race condition leading to two submissions receiving the same
+   colour balloon.
+ - Compile executables with `-std=gnu++20` now.
+ - Record the testcase directory for judging runs, this makes investigating
+   easier.
+ - Set default C/C++ standards in configure script, we intend to ship all
+   scripts free of warnings/errors.
+ - Add option to not show any rank on mirror contests, this is useful when
```
Comment (nickygerritsen) [inline ChangeLog:48]: What's the difference between this and line 58?

Comment (vmcj) [reply ChangeLog:48]: I would say nothing.

File: ChangeLog (hunk 7 above)
Comment (vmcj) [reply ChangeLog:51]: Yes, added a commit to hopefully make this more clear.

File: ChangeLog (hunk 6 above)
Comment (vmcj) [reply ChangeLog:54]: I'll remove it.

**Review by nickygerritsen — APPROVED** [review]

**Pull request merged**